src/app/static/src/app/utils.ts

```diff
@@ -348,3 +350,81 @@ export const getFormData = (file: File) => {
     formData.append("file", file)
     return formData
 }
+
+const transformPathToHash = (dataset: any) => {
+    Object.keys(dataset).map((key: string) => {
+        dataset[key] = {
+            hash: dataset[key].path.split("/")[1] || "",
+            filename: dataset[key].filename
+        }
+    })
+    return dataset
+}
+
+export const constructRehydrateProjectState = (rootState: RootState, data: ProjectRehydrateResultResponse) => {
+    const files = transformPathToHash({...data.state.datasets});
+
+    const modelCalibrate = {
+        ...rootState.modelCalibrate,
+        calibrateId: data.state.calibrate.id,
+        options: data.state.calibrate.options,
+        ready: true
+    }
+
+    const modelRun = {
+        ...rootState.modelRun,
+        modelRunId: data.state.model_fit.id,
+        ready: true
+    }
+
+    const modelOptions = {
+        ...rootState.modelOptions,
+        options: data.state.model_fit.options,
+    }
```

Comment:
I think these are outside the scope of this ticket - this was just about loading input files?